CONTRIBUTING.md

```diff
@@ -0,0 +1,184 @@
+# The Developer Workflow 
+
+-   [General Notes](#general-notes)
+-   [Issuing a Pull Request](#issuing-a-pull-request)
+-   [Reviewing a Pull Request](#reviewing-a-pull-request)
+-   [Running Tests](#running-tests)
+-   [Cautions](#cautions)
+-   [An Example](#an-example)
+    -   [Introduction](#introduction)
+    -   [Acquiring Moltres and Workflow](#acquiring-moltres-and-workflow)
+    -   [See also](#see-also)
+-   [Releases](#releases)
+
+------------------------------------------------------------------------
+
+## General Notes
+
+-   The terminology we use is based on the
+    [Integrator Workflow ](http://en.wikipedia.org/wiki/Integrator_workflow).
+
+-   Use a branching workflow similar to the one described
+    in the [progit gitbook](http://progit.org/book/ch3-4.html).
+
+-   Keep your own "master" and "develop" branches in sync with the
```

Comment:
I use `devel` after MOOSE. I'm ok with either one as long as we're consistent. So either we need to find and replace `develop` with `devel` in this document, or I need to rename the github `devel` branch `develop`, which would probably mean I need to get Brian Alger to reconfigure CIVET